src/directions/events.ts

```diff
@@ -116,7 +125,11 @@ export interface MapLibreGlDirectionsWaypointEventData {
 }
 
 export class MapLibreGlDirectionsWaypointEvent
-  implements MapLibreGlDirectionsEvent<MapMouseEvent | MapTouchEvent | undefined>
+  implements
+    MapLibreGlDirectionsEvent<
+      MapMouseEvent | MapTouchEvent | undefined,
+      "setwaypoints" | "rotatewaypoints" | "addwaypoint" | "removewaypoint" | "movewaypoint"
```

Comment:
Why's here `undefined`? Why the possible events are listed explicitly?

Reply:
Undefined was there previously. Events are passed in order for the `type` property to have precise type.